### ISSUE TITLE
[Feature] New `global_buf_path_line_sorted` trail mark mode

### ISSUE THREAD

**Issue by pidgeon777** [LeonHeidelbach/trailblazer.nvim#23]
### Did you check the docs?

- [X] I have read all the trailblazer.nvim docs

### Is your feature request related to a problem? Please describe.

Actually, when navigating through marks, it could be hard to figure out how a wished destination is far away.

In the current implementation of `global_buf_line_sorted`, it is possible to navigate marks by sorting buffer IDs.

This could be an example of a trail mark list for such a mode:

```
/path/to/project1/file1:10
/path/to/project1/file2:20
/path/to/project2/file1:30
/path/to/project1/file2:40
/path/to/project2/file2:50
```

But in many cases, it could be needed to switch to a particular location of a known project, and with the `global_buf_line_sorted` mode jumping can occur multiple times between two projects, before reaching the wished destination.

### Describe the solution you'd like

It would be great to add a mode, for example, `global_buf_path_line_sorted`, which sorts marks by buffer paths, and then by buffer lines.

Example of trail mark panel for such mode:

```
/path/to/project1/file1:10
/path/to/project1/file2:20
/path/to/project1/file2:40
/path/to/project2/file1:30
/path/to/project2/file2:50
```

Why paths? Because projects most of the time are set by paths, a trail mark mode which sorts marks by paths could be more useful than sorting by buffer IDs.

### Describe alternatives you've considered

N/A

**Comment by LeonHeidelbach** [comment]
@pidgeon777 Sounds reasonable and is a good idea. I think I will make the term "_buf_" interchangable with "_path_" in all available modes so you can decide yourself what sorting type you prefer. So `global_buf_line_sorted` would then work as well as `global_path_line_sorted`. :+1:

**Comment by pidgeon777** [comment]
Sounds good to me 👍.

**Comment by LeonHeidelbach** [comment]
@pidgeon777 This has been implemented here ba7928b12560b5497a5ab7c6d3617a8f84be5e13.

If you would like even more control over how trail marks are ordered within youre trail mark stacks you can now also use custom ordering added in one of the last commits. Simply use [these quickfix key maps](https://github.com/LeonHeidelbach/trailblazer.nvim#trail-mark-quickfix-list) to shift one or more trail marks up or down the stack which creates a new custom order selection mode. You can have individual custom order selections for each of your trail mark stacks. I currently use this feature like this:

1. I have one trail mark stack called "Files" in the custom order selection mode which contains only one mark per file for each file I frequently visit.
2. I have another trail mark stack called "Marks" that actually contains all trail marks for all files but use this in the `buffer_local_line_sorted` selection mode.

Now whenever I want to jump to a specific file, I simply switch stacks, select the file I want to move to and then switch back to my "Marks" stack. Now I have all marks specific to that file available to me. You could do the same thing on a per project basis and sort the marks just as you need them.
